### PR TITLE
Create feature toggles service auth

### PR DESCRIPTION
**Pull request by Luiz-FS**
**Feature/Bug description:** Create a login for the features management service.

**Solution:** I created the login function on the system.

**TODO/FIXME:** n/a

### CODE REVIEW

File: feature-toggles/user/user.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+function User(data) {
```
Comment (JuliePessoa) [inline feature-toggles/user/user.js]: why not using factory?

File: feature-toggles/utils/httpService.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+(function () {
+    'use strict';
+    
+    var app = angular.module('app');
+
+    app.service('HttpService', function HttpService($http) {
+        var service = this;
```
Comment (JuliePessoa) [inline feature-toggles/utils/httpService.js]: let/const for all

File: feature-toggles/app.js [hunk 3]
```diff
@@ -24,8 +26,8 @@
                 url: "/signin",
```
Comment (JuliePessoa) [inline feature-toggles/app.js]: sing in

File: feature-toggles/auth/authService.js [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                });
+            }).then(function(user) {
+                if (user.emailVerified) {
+                    return user.getIdToken(true).then(function(idToken) {
+                        return service.setupUser(idToken, user.emailVerified).then(function success(userInfo) {
+                            return userInfo;
+                        });
+                    });
+                } else {
+                    service.sendEmailVerification(user);
```
Comment (brobsc) [inline feature-toggles/auth/authService.js]: sendEmailVerification is undefined.

File: feature-toggles/auth/loginController.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            var promise = AuthService.loginWithGoogle();
+            promise.then(function success() {
+                redirectTo(redirectPath);
+            }).catch(function(error) {
+                loginCtrl.showToast(error);
+            });
+            return promise;
+        };
+
+        loginCtrl.limpar = function limpar() {
```
Comment (brobsc) [inline feature-toggles/auth/loginController.js]: unused function.

File: feature-toggles/user/user.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    data = data || {};
+    _.extend(this, data);
+
+    if (!this.current_institution) {
+        this.changeInstitution();
+    }
+}
+
+User.prototype.changeInstitution = function changeInstitution(institution) {
```
Comment (brobsc) [inline feature-toggles/user/user.js]: Will this be used?

File: feature-toggles/auth/loginController.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+        function redirectTo(path) {
+            if (path) {
+                window.location.pathname = path;
+            } else {
+                $state.go("manage-features");
+            }
+        }
+
+        (function main() {
```
Comment (tiagolimpe) [inline feature-toggles/auth/loginController.js]: Change to $onInit

Comment (tiagolimpe) [inline feature-toggles/auth/loginController.js]: Change to $onInit

File: feature-toggles/user/user.js (hunk 6 above)
Comment (Luiz-FS) [reply feature-toggles/user/user.js]: Yes, it is being used when creating the user, line 8.

**Pull request merged**